### ISSUE TITLE
How can I use ogv just for decoding?

### ISSUE THREAD

**Issue by jfrancos** [brion/ogv.js#533]
For Brave/Firefox I use WebAudio's [`decodeAudioData`](https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/decodeAudioData), which returns an [`AudioBuffer`](https://developer.mozilla.org/en-US/docs/Web/API/AudioBuffer) of the decoded PCM audio data.  This doesn't work in Safari.

Is there a way I can use ogv just for decoding, in place of `decodeAudioData`?

Thanks for your hard work.

**Comment by brion** [comment]
This would be possible to rig up using the codec modules directly, though not yet a convenient interface for it.

Roughly, you need to pass data into a demuxer module's `receiveInput` method, then loop over it with `process`, dequeuing any packets along the way and passing them into the decoder module. Take the output data, concatenate it all into a single ArrayBuffer and resolve the promise.

Devil's always in the details though. :)

**Comment by guest271314** [comment]
What is input and expected output?

**Comment by guest271314** [comment]
Do not have access to Safari. Testing `decodeAudioData` at Epiphany 3.28.5 an `AudioBuffer` is logged when the audio codec is `Ogg` or `A_PCM/FLOAT/IEEE` in a WebM container or `MP2/MP3` in a `MP3`. `OPUS` in a WebM container outputs `null`. Conversion to `WAV` can be achieved using this version https://cdn.jsdelivr.net/npm/audiobuffer-to-wav@1.0.0/index.js of https://github.com/mattdiamond/Recorderjs at https://plnkr.co/edit/HMqf5CrAXhOHtlzOadI8?p=info.

**Comment by jfrancos** [comment]
Actually I figured out how to use emscripten to be able to use vorbis-tools' oggdec directly:

https://github.com/jfrancos/oggdec-wasm

**Comment by guest271314** [comment]
@jfrancos Are you able to verify which file types and codecs Safari is able to decode with `decodeAudioData()` re https://github.com/brion/ogv.js/issues/533#issuecomment-565857607 and https://github.com/mdn/webaudio-examples/issues/5?

**Comment by rgov** [comment]
I would love to be able to do this as well. My use case is that I have a stream of "raw" Theora packets coming from a camera, without an OGV container. I want to be able to feed in the header packets, an intra frame, and a sequence of inter frames and get out the final image buffer. Is this possible at all? The `codec-bench.html` file looks like it might be close.

**Comment by brion** [comment]
Yes, you can call the decoder directly if you like. No stable api or documentation or support.

**Comment by rgov** [comment]
Ok thanks. I put together an example of using ogv.js to decode raw Theora packets into video frames: https://github.com/rgov/js-theora-decoder ([Demo](https://rgov.github.io/js-theora-decoder/))
